curp/src/server/raw_curp/mod.rs

```diff
@@ -0,0 +1,806 @@
+//! READ THIS BEFORE YOU START WRITING CODE FOR THIS MODULE
+//! To avoid deadlock, let's make some rules:
+//! 1. To group similar functions, I divide Curp impl into three scope: one for utils(don't grab lock here), one for tick, one for handlers
+//! 2. Lock order should be:
+//!     1. self.st
+//!     2. self.ctx || self.ltx (there is no need for grabbing both)
+//!     3. self.log
+
+#![allow(clippy::similar_names)] // st, lst, cst is similar but not confusing
+#![allow(clippy::indexing_slicing)] // use index to access log is safe
+#![allow(clippy::integer_arithmetic)] // u64 is large enough and won't overflow
```

Comment:
I think it's fine because I have done some math:
Even if we have 1 million entries per second, it would take 584942 years to overflow, which could somehow be regarded as impossible.

Reply:
Then we should comment the integer arithmetic when it possible overflow and explain the reason.